core/src/autogluon/core/models/ensemble/bagged_ensemble_model.py

```diff
@@ -464,12 +453,14 @@ def _fit_folds(self,
             fold_fitting_strategy = self._get_default_fold_fitting_strategy()
         num_folds_parallel = self.params.get('num_folds_parallel', 'auto')
         disable_parallel_fitting = self.params.get('_disable_parallel_fitting', False)
-        if fold_fitting_strategy == 'parallel_local':
+        if fold_fitting_strategy in ['parallel_local', "parallel_distributed"]:
+            if fold_fitting_strategy == "parallel_local":
+                fold_fitting_strategy = ParallelLocalFoldFittingStrategy
+            else:
+                fold_fitting_strategy = ParallelDistributedFoldFittingStrategy
```

Comment:
This logic is getting sufficiently complicated that it deserves its own method. All the logic prior to L472 should be moved to its own method.